syntax = "proto3";
package tino.daemon.v1;

service DaemonService {
  rpc GetSystemInfo(GetSystemInfoRequest) returns (GetSystemInfoResponse);
  rpc Shutdown(ShutdownRequest) returns (ShutdownResponse);
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

message GetSystemInfoRequest {}
message GetSystemInfoResponse {
  string python_version = 1;
  string nautilus_version = 2;
  uint64 memory_usage_bytes = 3;
  string uptime = 4;
}

message ShutdownRequest {}
message ShutdownResponse {
  bool success = 1;
}

message HealthCheckRequest {}
message HealthCheckResponse {
  bool healthy = 1;
  string uptime = 2;
  string version = 3;
  repeated string connected_exchanges = 4;
}
